src/watch/fileWatchers.ts

```diff
@@ -60,9 +60,9 @@ export default class FileWatcher {
 
 		const handleWatchEvent = (event: string) => {
 			if (event === 'rename' || event === 'unlink') {
-				this.close();
-				group.delete(id);
+				modifiedTime = -1;
```

Comment:
Not strictly related to the issue at hand but I noticed that when I delete and recreate a file that is required for a build to succeed, the recreation will not trigger a new run, which is bad. Now the only time watched files are removed is when a run was successful; otherwise we just accumulate watched files so that we do not miss anything important.

Reply:
Ok, we can do this for chokidar but it does not seem to work well for fs.watch...

Reply:
Ok, running chokidar still fails on CI. I guess I'll just have to revert this.

Reply:
I assume it depends on the operating system if that works as chokidar will use different techniques. I guess we will need to live with this, you can still trigger another run by modifying any other file that was already part of the graph.